apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -413,6 +417,14 @@ type Image struct {
 	DiskFormat *string `json:"format,omitempty"`
 }
 
+// Custom deploy is a description of a customized deploy process.
+type CustomDeploy struct {
+	// Custom deploy method name.
+	// This name is specific to the deploy ramdisk used. If you don't have
+	// a custom deploy ramdisk, you shouldn't use CustomDeploy.
+	Method string `json:"method"`
```

Comment:
What about args?

Reply:
I was requested to omit args on the review of the design document. They can be added later if we have a good case (and an idea how to wire them in the API).

Reply:
What do you mean `a good case`? Passing args to the method isn't a good case?

Also `Args` is required field in DeployStep https://github.com/gophercloud/gophercloud/blob/v0.17.0/openstack/baremetal/v1/nodes/requests.go#L427

Reply:
This implementation passes empty args (btw I'm not sure why we require them in ironic).

Reply:
By a good case I mean a use case for passing args. What I'm doing does not require non-empty args.

Reply:
E.g when someone want to use ansible to deploy.

Your comment:

> #180 has been made much more generic, and you may be able to reuse its interface for your goals. E.g. you allow `CustomDeploy.Method = "ansible"` to designate the ansible deploy. Then you can use metaData for passing additional information.

https://github.com/metal3-io/metal3-docs/pull/181#discussion_r629368593

But regardless this comment, passing args to methods is pretty standard approach and methods without args are pretty much useless.

I understand that doc is merged and only `Method` is mentioned there, but am I only one who confused here?

"Ironic api allows to pass args but let's not use it."

I also can't see any discussion about removing args here: https://github.com/metal3-io/metal3-docs/pull/180

What is wrong with args?

Reply:
The previous discussion is https://github.com/metal3-io/metal3-docs/pull/180#discussion_r626924686. The problem was building API that accepts arbitrary JSON data. We already have `MetaData` and `UserData` for that, although it may not work for the ansible case. In any case, I suggest you rebase your older design on top of https://github.com/metal3-io/metal3-docs/pull/180 and discuss it there. This patch implements the approved design.